src/test/java/com/machowina/service/TicketServiceTest.java

```diff
@@ -0,0 +1,172 @@
+	private User driver;
+	private ParkingZone zone;
+
+	@Before
+	public void setUp() throws Exception {
+		
+		ticketService = new TicketServiceImp(carService, zoneService, ticketRepository, parkingRatesService);
+	
+		driver = new User("kowalski","pass","regular");
+		car = new Car("WI99021", driver);
+		zone = new ParkingZone("Warszawa","Strefa płatnego parkowania");
+	}
+	
+	
+	
+	@Test
+	public void testStopTicket() {
+		//given
+		ParkingTicket ticket  = new ParkingTicket(LocalDateTime.now(), zone, car, driver);
+		Mockito.when(ticketRepository.findOne(1l)).thenReturn(ticket);
+		Mockito.when(ticketRepository.save(ticket)).thenReturn(ticket);
+		
+		//when
+		ticketService.stopTicket(1l);
+		//then
+		Assert.assertTrue(ticket.isStopped());
+		Assert.assertNotNull(ticket.getStopTime());
+	}
+	
+	@Test(expected = TicketAlreadyStopped.class)
+	public void testStopTicket_already_stopped() {
+		//given
+		ParkingTicket ticket  = new ParkingTicket(LocalDateTime.now(), zone, car, driver);
+		ticket.setStopped(true);
+		Mockito.when(ticketRepository.findOne(1l)).thenReturn(ticket);
+		//when
+		ticketService.stopTicket(1l);
+		//then catch exception
+	}
+	
+	@Test
+	public void testFindById() {
+		//given
+		ParkingTicket ticket  = new ParkingTicket(LocalDateTime.now(), zone, car, driver);
+		Mockito.when(ticketRepository.findOne(1l)).thenReturn(ticket);
+		//when
+		ParkingTicket foundTicket = ticketService.findById(1l);
+		//then
+		Assert.assertEquals(car, foundTicket.getCar());
+		Assert.assertEquals(driver, foundTicket.getDriver());
+		Assert.assertEquals(zone, foundTicket.getParkingZone());
+	}
+	
+	@Test(expected = EntityNotFoundException.class)
+	public void testFindById_not_found() {
+		//given
+		Mockito.when(ticketRepository.findOne(1l)).thenReturn(null);
+		//when
+		ParkingTicket foundTicket = ticketService.findById(1l);
+		//then catch exception
+	}
+	
+	@Test
+	public void testCreateTicket() {
+		//given
+		Mockito.when(carService.findById(1l)).thenReturn(car);
+		Mockito.when(zoneService.findOne(1l)).thenReturn(zone);
+		//when
+		ParkingTicket newTicket = ticketService.createTicket(1l, 1l);
+		//then
+		Assert.assertEquals(car, newTicket.getCar());
+		Assert.assertEquals(driver, newTicket.getDriver());
+		Assert.assertEquals(zone, newTicket.getParkingZone());
+		Assert.assertTrue(LocalDateTime.now().plusSeconds(1l).isAfter(newTicket.getStartTime()));
+		Assert.assertEquals(LocalDateTime.now().getMinute(), newTicket.getStartTime().getMinute());
+		Assert.assertFalse(newTicket.isStopped());
+		Assert.assertFalse(newTicket.isPaid());
+		Assert.assertNull(newTicket.getStopTime());
+		
+
+	}
+	
+	@Test(expected = EntityNotFoundException.class)
+	public void testCreateTicket_with_no_car() {
+		//given
+		Mockito.when(carService.findById(1l)).thenThrow(EntityNotFoundException.class);
+		Mockito.when(zoneService.findOne(1l)).thenReturn(zone);
+		//when
+		ParkingTicket newTicket = ticketService.createTicket(1l, 1l);
+		//then
```

Comment:
gdzie indziej pisałaś `then catch exception` - po co komentarze, które nie mówią prawdy albo są niespójne?

Reply:
Zapomniałam tu o catch exception - ale te komentarze pomagają w oddzieleniu części testu od siebie i myślę że wygląda to czytelniej